Modules/_io/fileio.c

```diff
@@ -66,6 +66,7 @@ typedef struct {
     unsigned int closefd : 1;
     char finalizing;
     unsigned int blksize;
+    __int64_t size;
```

Comment:
```suggestion
    Py_ssize_t size;
```

Reply:
It should be `off_t`. The file size can exceed the `Py_ssize_t` range.

Reply:
You are right! size handling depends on platform and support for large file support. You can copy the approach from ``posixmodule.c``:

```
#ifdef MS_WINDOWS
    typedef long long Py_off_t;
#else
    typedef off_t Py_off_t;
#endif
```

```
#ifdef HAVE_LARGEFILE_SUPPORT
    size = (Py_off_t)PyLong_AsLongLong(arg);
#else
    size = (Py_off_t)PyLong_AsLong(arg);
#endif
```

Reply:
It is too complicated, we need to duplicate this in several files. But it may be simpler if only cache the boolean result (`st_size == 0` or `S_IFCHR()`) as `_maybeatty`. Then the size of `off_t` does not matter.